feat(theme): theme prop is now available on each component (#13718)
vuetifyjs/vuetify: packages/docs/src/pages/en/components/theme-providers.md

---
meta:
  title: Theme provider component
  description: The theme provider allows you to style a section of your application in a different theme from the default
  keywords: theme provider, vuetify theme provider component, vue theme provider component
related:
  - /features/theme/
---

# Theme providers

The theme provider allows you to style a section of your application in a different theme from the default

## API

- [v-theme-provider](/api/v-theme-provider)

## Examples

### Background

> By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.

Author: 1. The link for renderless component goes to page on render functions, which does not contain any specific information about the concept of renderless components
2. Is it relevant for the user that theme-provider uses provide/inject?

> By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.

Reviewer: 1. Thought they were the same. We can update the page link or remove it entirely if it doesn't make sense
2. It introduces the user to a concept we use throughout Vuetify 3
3. Don't understand but if it's just a language update, what should it change to?

> By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.

Author: 1. I'd say remove. I don't think official vue docs talk about renderless anywhere
2. Not sure if this is the place to do it
3. `When you use the with-background prop the v-theme-provider will render a div and apply the background color or something something something`

> By default `v-theme-provider` will act as a renderless component, and simply provide a new theme context to all of its children. However if you use the **with-background** prop, it will render a `div` and apply the correct background color to it according to the theme used.

Reviewer: 1. 👍 
2. Okay, I'll defer to your judgement
3. `When using the **with-background** prop, the `v-theme-provider` renders a `<div>` element and applies the specified theme's background color to the element.

### Background

> By default, `v-theme-provider` is a renderless component that allows you to change the applied theme for all of its children. When using the **with-background** prop, the `v-theme-provider` will wrap its children in an element and apply the correct background color to it according to the theme used.

Reviewer: ```suggestion
By default, `v-theme-provider` is a renderless component that allows you to change the applied theme for all of its children. When using the **with-background** prop, the `v-theme-provider` wraps its children in a `<div>` element and applies selected **theme's** background color.
```
